crates/polars-io/src/cloud/glob.rs

```diff
@@ -178,16 +178,25 @@ pub async fn glob(url: &str, cloud_options: Option<&CloudOptions>) -> PolarsResu
         },
         store,
     ) = super::build_object_store(url, cloud_options).await?;
-    let matcher = Matcher::new(prefix.clone(), expansion.as_deref())?;
+    let matcher = Matcher::new(
+        if scheme == "file" {
+            // The returned location has the leading slash stripped.
+            prefix[1..].to_string()
+        } else {
+            prefix.clone()
+        },
+        expansion.as_deref(),
+    )?;
 
     let list_stream = store
         .list(Some(&Path::from(prefix)))
         .map_err(to_compute_err);
-    let locations: Vec<Path> = list_stream
+    let mut locations: Vec<Path> = list_stream
         .then(|entry| async { Ok::<_, PolarsError>(entry.map_err(to_compute_err)?.location) })
         .filter(|name| ready(name.as_ref().map_or(true, |name| matcher.is_matching(name))))
         .try_collect()
         .await?;
+    locations.sort_unstable();
```

Comment:
We did not previously use this function `async fn glob` in our test suite even with FORCE_ASYNC, but in this PR we do, which led to lots of test failures as the files were not read in a consistent order as the file list wasn't sorted.